lld/ELF/ScriptParser.cpp

```diff
@@ -611,6 +613,34 @@ SmallVector<SectionCommand *, 0> ScriptParser::readOverlay() {
   return v;
 }
 
+SectionClassDesc *ScriptParser::readSectionClassDescription() {
+  StringRef name = readSectionClassName();
+  SectionClassDesc *desc = make<SectionClassDesc>(name);
+  if (!script->sectionClasses.insert({CachedHashStringRef(name), desc}).second)
+    setError("section class '" + name + "' already defined");
+  expect("{");
+  while (!errorCount() && !consume("}")) {
```

Comment:
With #100493  , the loop pattern needs a change.

Reply:
Hmm, I had used `readOutputSectionDescription` as a reference for the parsing idioms, but that didn't change in #100493 . Is there a substantive difference between that and `readSectionClassDescription`, or was the loop in `readOutputSectionDescription` missed? It appears that there are still 15 instances of the `while(!errorCount() && !consume(...` idiom in ScriptParser.cpp, but it isn't obvious to me what the differences are between those changed and those that remain.

Reply:
I neglected `while (... consume("}"))` in #100493 , which might lead to bad diagnostics or dead loops.

I introduced `till()` in 2a89356d64a274f7171b45d8b7b05b6315a6f627 and refactored some `while (... consume("}"))` to use `till`. Use that instead?

Reply:
Done.